caas/kubernetes/provider/resources/applier.go

```diff
@@ -52,29 +56,73 @@ func (op *operation) process(ctx context.Context, rollback Applier) error {
 	} else if err != nil {
 		return errors.Annotatef(err, "checking if resource %q exists or not", existingRes.ID())
 	}
-	if found {
-		ver := op.resource.GetObjectMeta().GetResourceVersion()
-		if ver != "" && ver != existingRes.GetObjectMeta().GetResourceVersion() {
-			id := op.resource.ID()
-			return errors.Annotatef(errConflict, "%s %s", id.Type, id.Name)
-		}
-	}
+
 	switch op.opType {
 	case opApply:
-		err = op.resource.Apply(ctx)
-		if found {
-			// apply the previously existing resource.
-			rollback.Apply(existingRes)
-		} else {
-			// delete the new resource just created.
-			rollback.Delete(op.resource)
+		err = retry.Call(retry.CallArgs{
+			Func: func() error {
+				err := op.resource.Apply(ctx)
+				if errors.Is(err, errConflict) {
+					_ = op.resource.Get(ctx) // refresh resource version
```

Comment:
I'm not sure we should be ignoring the error here.
eg it could be not found

Reply:
add comment as to why we ignore err